src/Data/Json/Extended/Signature/Json.purs

```diff
@@ -3,26 +3,25 @@ module Data.Json.Extended.Signature.Json where
 import Prelude
 
 import Control.Alt ((<|>))
-
 import Data.Argonaut.Core as JS
 import Data.Argonaut.Decode (class DecodeJson, decodeJson, (.?))
 import Data.Argonaut.Encode (encodeJson)
 import Data.Bifunctor (lmap)
 import Data.Either as E
+import Data.HugeInt as HI
 import Data.HugeNum as HN
 import Data.Int as Int
 import Data.Json.Extended.Signature.Core (EJsonF(..), EJsonMap(..))
 import Data.Maybe as M
 import Data.StrMap as SM
 import Data.Traversable as TR
-
 import Matryoshka (Algebra, CoalgebraM)
 
 encodeJsonEJsonF ∷ Algebra EJsonF JS.Json
 encodeJsonEJsonF = case _ of
   Null → JS.jsonNull
   Boolean b → encodeJson b
-  Integer i → encodeJson i
+  Integer i → encodeJson $ HN.toNumber $ HI.toHugeNum i -- TODO: bug in HI.toInt
```

Comment:
We do now :smile: https://github.com/purescript-contrib/purescript-precise/issues/10

My WIP stuff on `-precise` will probably fix it anyway, but you're right it's worth having the issue.